packages/node_modules/@node-red/runtime/lib/module/index.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+const fs = require("fs-extra");
+const path = require("path");
+const crypt = require("crypto-js");
+const mustache = require("mustache");
+const util = require("@node-red/util");
+const exec = util.exec;
+
+const NPM_COMMAND = (process.platform === "win32") ? "npm.cmd" : "npm";
+
+let runtime = null;
+let settings = null;
+
+
+function getTempDir() {
+    const dir = (settings.userDir || process.env.NODE_RED_HOME || ".");
```

Comment:
Use `os.tmpdir()` rather than fill the userDir. That ensures the directory is deleted - something I don't see happening anywhere with this code.

Reply:
Fixed to use `os.tmpdir`.